Mathlib/AlgebraicGeometry/EllipticCurve/Jacobian.lean

```diff
@@ -285,39 +289,39 @@ lemma eval_polynomialY_of_Z_ne_zero {P : Fin 3 → F} (hPz : P z ≠ 0) :
   linear_combination (norm := (rw [eval_polynomialY, Affine.eval_polynomialY]; ring1))
     W.a₁ * P x / P z ^ 2 * div_self hPz + W.a₃ * div_self (pow_ne_zero 3 hPz)
 
-variable (V) in
+variable (W') in
 /-- The partial derivative $W_Z(X, Y, Z)$ of $W(X, Y, Z)$ with respect to $Z$. -/
 noncomputable def polynomialZ : MvPolynomial (Fin 3) R :=
-  pderiv z V.polynomial
+  pderiv z W'.polynomial
 
-lemma polynomialZ_eq : V.polynomialZ =
-    C V.a₁ * X 0 * X 1 + C (3 * V.a₃) * X 1 * X 2 ^ 2
-      - (C (2 * V.a₂) * X 0 ^ 2 * X 2 + C (4 * V.a₄) * X 0 * X 2 ^ 3 + C (6 * V.a₆) * X 2 ^ 5) := by
+lemma polynomialZ_eq : W'.polynomialZ = C W'.a₁ * X 0 * X 1 + C (3 * W'.a₃) * X 1 * X 2 ^ 2 -
+    (C (2 * W'.a₂) * X 0 ^ 2 * X 2 + C (4 * W'.a₄) * X 0 * X 2 ^ 3 + C (6 * W'.a₆) * X 2 ^ 5) := by
   rw [polynomialZ, polynomial]
   pderiv_simp
   ring1
 
-lemma eval_polynomialZ (P : Fin 3 → R) : eval P V.polynomialZ =
-    V.a₁ * P x * P y + 3 * V.a₃ * P y * P z ^ 2
-      - (2 * V.a₂ * P x ^ 2 * P z + 4 * V.a₄ * P x * P z ^ 3 + 6 * V.a₆ * P z ^ 5) := by
+lemma eval_polynomialZ (P : Fin 3 → R) : eval P W'.polynomialZ =
+    W'.a₁ * P x * P y + 3 * W'.a₃ * P y * P z ^ 2 -
+      (2 * W'.a₂ * P x ^ 2 * P z + 4 * W'.a₄ * P x * P z ^ 3 + 6 * W'.a₆ * P z ^ 5) := by
   rw [polynomialZ_eq]
   eval_simp
 
-variable (V) in
-/-- The proposition that a point representative $(x, y, z)$ in `W` is nonsingular.
+variable (W') in
+/-- The proposition that a point representative $(x, y, z)$ in `W'` is nonsingular.
```

Comment:
You might want to make the remark that this definition is not mathematically sensible if R isn't a field (even though it compiles). I think that in the non-field case the correct definition is probably that the ideal generated by the three evals is all of R. Otherwise you can specialise at a maximal ideal containing the ideal generated by the evals and get a singular specialization.